src/service/news.service.ts

```diff
@@ -20,19 +20,18 @@ export class NewsService {
   getNewsByCoordinates(
     latitude: number,
     longitude: number,
+    localOnly: boolean,
     blacklistSources: string,
   ): Observable<NewsData[]> {
-    return this.http.get<NewsData[]>(
-      `${this.apiUrl}?lon=${longitude}&lat=${latitude}&blacklistSources=${blacklistSources}`,
-    );
-  }
+    let params = new HttpParams();
 
-  getNewsCoordinats(
-    latitude: number,
-    longitude: number,
-    blacklistSources: any,
-  ): Observable<any> {
-    const url = `${this.apiUrl}?lon=${longitude}&lat=${latitude}&blacklistSources=${blacklistSources}`;
-    return this.http.get(url);
+    if (localOnly) {
+      params = params.set('localOnly', 'true');
+    }
+    if (blacklistSources) {
+      params = params.set('blacklistSources', blacklistSources);
+    }
+    const url = `${this.apiUrl}?lon=${longitude}&lat=${latitude}`;
```

Comment:
We should do all the params the same way for consistency. And also to eliminate the possibility of getting a doubble `?` in the URL.